GCP subnet share conditions not working correctly

kopachevsky opened the pull request:
Issue  #152

Ready for review

aaron-lane commented:
Fixes #152.

aaron-lane commented on modules/core_project_factory/main.tf:
```diff
@@ -70,6 +70,28 @@ EOD
+  }
+}
+
+resource "null_resource" "check_if_shared_vpc_subnets_contais_items_with_invalid_name" {
```
```suggestion
resource "null_resource" "check_if_shared_vpc_subnets_contains_items_with_invalid_name" {
```

aaron-lane commented on test/integration/full/controls/shared-vpc.rb:
```diff
@@ -88,28 +88,41 @@
+    }
+  ]
+
+  SHARED_VPC_SUBNETS_DATA_SET.each do |subnet|
```
Test cases must not be generated through metaprogramming; this practice can lead to false positives if the logic driving the generation happens to change.

kopachevsky replied:
I agree, but how would you minimize code duplication in this case? basically we need to do same code against 2 different networks

aaron-lane replied:
Reducing code duplication in test cases is less of a priority than creating clear test cases. Duplicating logic in this scenario is totally acceptable.

kopachevsky commented:
@aaron-lane 
> Can you please add a CHANGELOG entry and provide a [gist](https://gist.github.com) which contains the output of a Terraform process when it fails due to an invalid subnet being passed to `shared_vpc_subnets`?

Added CHANGELOG
Regarding gist, it can take to much time execute locally full test without G Suite, so I've added bad sunet name in previous commit and got:
       
```
       1 error(s) occurred:
       
       * module.project-factory.module.project-factory.null_resource.check_if_shared_vpc_subnets_contains_items_with_invalid_name: Error running command 'false': exit status 1. Output: 
```
       
on Concourse  build 422, should I copy/paste full terraform output from there for Full test suite?  I think it could contain sensitive info
